lib/sevak/core.rb

```diff
@@ -1,15 +1,27 @@
 module Sevak
   module Core
     def connection
-      return @conn if @conn
-      @conn ||= Bunny.new(
-        host: config.host,
-        port: config.port,
-        username: config.user,
-        password: config.password)
-
-      @conn.start
-      @conn
+      attempt = 0
+      begin
+        if attempt < 10
+          host = config.host.kind_of?(Array) ? config.host : [config.host]
+          @conn = Bunny.new(
+            hosts: host,
+            port: config.port,
+            username: config.user,
+            password: config.password)
+          @conn.start
+          @conn
+        end
+      rescue Bunny::TCPConnectionFailedForAllHosts => e
+        attempt += 1
+        sleep(0.1)
```

Comment:
sleep(0.1) if attempt < 10 would be neat, or the main bock will be a a conditional